### ISSUE TITLE
translate-c static function variable shadowing other functions parameters 

### ISSUE THREAD

**Issue by lacebal** [ziglang/zig#8208]
**Zig version: 0.8.0-dev.1431+c760532be**

**Problem:  static function variable is shadowing other functions arguments with same name.** 

**Error message:** 
```
$ zig build -Dtarget=native-native-gnu
.\zig-cache\o\ea6804b44f6b8f259469a108221aed6a\cimport.zig:3:5: error: redefinition of 'size'
    var size = arg_size;
    ^
.\zig-cache\o\ea6804b44f6b8f259469a108221aed6a\cimport.zig:7:1: note: previous definition is here
var size: c_int = 0;
```

**Repro example**
https://gist.github.com/lacebal/a77a4cb24e0d738e1a2c4a977da0cf7f

**Same error was observed when const static var shadows functions params when integrating nuklear C library: https://github.com/Immediate-Mode-UI/Nuklear**

staticshadow.h
```
extern int paramsize(int);
extern int staticshadowing(void);

#ifdef STATICSHADOW_IMPL
extern int paramsize(int size) {
    size++;
    return size;
}
extern int staticshadowing(void) {
    static int size = 0;
    size++;
    return size;
}
#endif
```

main.zig
```
const c = @cImport({
    // @cDefine("NK_INCLUDE_FIXED_TYPES", {});
    // @cDefine("NK_INCLUDE_STANDARD_IO", {});
    // @cDefine("NK_INCLUDE_DEFAULT_ALLOCATOR", {});
    // @cDefine("NK_INCLUDE_VERTEX_BUFFER_OUTPUT", {});
    // // @cDefine("NK_INCLUDE_FONT_BAKING", {});
    // // @cDefine("NK_INCLUDE_DEFAULT_FONT", {});
    // @cDefine("NK_IMPLEMENTATION", {});
    // @cInclude("nuklear.h");

    @cDefine("STATICSHADOW_IMPL", {});
    @cInclude("staticshadow.h");
});
const std = @import("std");

pub fn main() void {
    // var cmds: c.struct_nk_buffer = c.struct_nk_buffer {
    //     .marker = [_]c.struct_nk_buffer_marker{ 
    //         c.struct_nk_buffer_marker{ .active = undefined, .offset = undefined } 
    //     } ** 2,
    //     .pool = undefined,
    //     .type = undefined,
    //     .memory = c.struct_nk_memory{ .ptr = undefined, .size = undefined },
    //     .grow_factor = undefined,
    //     .allocated = undefined,
    //     .needed = undefined,
    //     .calls = undefined,
    //     .size = undefined,
    // };
    // c.nk_buffer_init_default(&cmds);
    std.log.info("Size: {} {}", .{ c.paramsize(10), c.staticshadowing() });
}
```

build.zig
```
const Builder = @import("std").build.Builder;
const x86 = @import("std").Target.x86;

pub fn build(b: *Builder) void {
    // Standard target options allows the person running `zig build` to choose
    // what target to build for. Here we do not override the defaults, which
    // means any target is allowed, and the default is native. Other options
    // for restricting supported target set are available.
    var target = b.standardTargetOptions(.{});
    
    // Standard release options allow the person running `zig build` to select
    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
    const mode = b.standardReleaseOptions();

    const exe = b.addExecutable("app-zig-bug", "src/main.zig");
    exe.setBuildMode(mode);
    exe.setTarget(target);
    // nuklear
    exe.addSystemIncludeDir("vendor/nuklear/");
    // src
    exe.addSystemIncludeDir("src/");
    
    // C and win32
    exe.linkLibC();
    exe.install();

    const run_cmd = exe.run();
    run_cmd.step.dependOn(b.getInstallStep());
    if (b.args) |args| {
        run_cmd.addArgs(args);
    }

    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);
}
```

cimport.zig extract
```
pub usingnamespace @import("std").c.builtins;
pub fn paramsize(arg_size: c_int) callconv(.C) c_int {
    var size = arg_size;
    size += 1;
    return size;
}
var size: c_int = 0;
pub fn staticshadowing() callconv(.C) c_int {
    size += 1;
    return size;
}
...
```

**Comment by Vexu** [comment]
This can also happen with extern variables declared inside functions:
```c
void foo() {
    int a;
}
void bar() {
    extern int a;
}
```

**Comment by andrewrk** [comment]
Fixed in ea4a25287eb65e639fd75d425f320663e6e8dca1.
